dspace-api/src/main/java/org/dspace/content/RelationshipServiceImpl.java

```diff
@@ -0,0 +1,398 @@
+    }
+
+    private boolean isRelationshipValidToCreate(Context context, Relationship relationship) throws SQLException {
+        RelationshipType relationshipType = relationship.getRelationshipType();
+
+        if (!verifyEntityTypes(relationship.getLeftItem(), relationshipType.getLeftType())) {
+            log.warn("The relationship has been deemed invalid since the leftItem" +
+                         " and leftType do no match on entityType");
+            logRelationshipTypeDetails(relationshipType);
+            return false;
+        }
+        if (!verifyEntityTypes(relationship.getRightItem(), relationshipType.getRightType())) {
+            log.warn("The relationship has been deemed invalid since the rightItem" +
+                         " and rightType do no match on entityType");
+            logRelationshipTypeDetails(relationshipType);
+            return false;
+        }
+        if (!verifyMaxCardinality(context, relationship.getLeftItem(),
+                                  relationshipType.getLeftMaxCardinality(), relationshipType)) {
+            log.warn("The relationship has been deemed invalid since the left item has more" +
+                         " relationships than the left max cardinality allows after we'd store this relationship");
+            logRelationshipTypeDetails(relationshipType);
+            return false;
+        }
+        if (!verifyMaxCardinality(context, relationship.getRightItem(),
+                                  relationshipType.getRightMaxCardinality(), relationshipType)) {
+            log.warn("The relationship has been deemed invalid since the right item has more" +
+                         " relationships than the right max cardinality allows after we'd store this relationship");
+            logRelationshipTypeDetails(relationshipType);
+            return false;
+        }
+        return true;
+    }
+
+    private void logRelationshipTypeDetails(RelationshipType relationshipType) {
+        log.warn("The relationshipType's ID is: " + relationshipType.getID());
+        log.warn("The relationshipType's left label is: " + relationshipType.getLeftLabel());
+        log.warn("The relationshipType's right label is: " + relationshipType.getRightLabel());
+        log.warn("The relationshipType's left entityType label is: " + relationshipType.getLeftType().getLabel());
+        log.warn("The relationshipType's right entityType label is: " + relationshipType.getRightType().getLabel());
+        log.warn("The relationshipType's left min cardinality is: " + relationshipType.getLeftMinCardinality());
+        log.warn("The relationshipType's left max cardinality is: " + relationshipType.getLeftMaxCardinality());
+        log.warn("The relationshipType's right min cardinality is: " + relationshipType.getRightMinCardinality());
+        log.warn("The relationshipType's right max cardinality is: " + relationshipType.getRightMaxCardinality());
```

Comment:
I think the method name is not clear enough. This method is only used when in incorrect action is attempted, and the data is required to identify the problem.
We'll rename the method to make it clear

Reply:
I see that before each `logRelationshipTypeDetails` call, there is already a warning message giving more information. I still think that logging all relationship's type details is useful when debugging a problem, but that it would create too much log spam when actually running in production.

Reply:
There's an initial brief mention at https://github.com/DSpace/DSpace/blob/configurable_entities/dspace-api/src/main/java/org/dspace/content/RelationshipServiceImpl.java#L193 with the full details at https://github.com/DSpace/DSpace/blob/configurable_entities/dspace-api/src/main/java/org/dspace/content/RelationshipServiceImpl.java#L222

This is actually comparable with adding a metadata value to an item which doesn't exist. In that case, it's even displayed as an ERROR in the logs.
The user is trying to perform an action which is not authorized in the current config, so it's important to see the details of the action, so it can be verified what they were trying to do.